internal/ingress/controller/controller_test.go

```diff
@@ -256,6 +256,28 @@ func TestCheckIngress(t *testing.T) {
 			}
 		})
 
+		t.Run("When a new catch-all ingress is being created despite catch-alls being disabled ", func(t *testing.T) {
+			nginx.command = testNginxTestCommand{
+				t:   t,
+				err: nil,
+			}
+			nginx.cfg.DisableCatchAll = true
+
+			ing.Spec.Backend = &networking.IngressBackend{
+				ServiceName: "http-svc",
+				ServicePort: intstr.IntOrString{
+					IntVal: 80,
+				},
+			}
+
+			if nginx.CheckIngress(ing) == nil {
+				t.Errorf("with a new catch-all ingress and catch-alls disable, should return error")
+			}
+
+			// set back to nil for next test
+			ing.Spec.Backend = nil
```

Comment:
This assumes Backend will always be `nil` before the line 266. Instead we should take the original value before like 266 and here restore the original value whatever it was

Reply:
oh yeah your right I'll fix that